### PR TITLE
Problem: abci handshake is not shutdown gracefully

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- use cancellable node constructor

Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (ante) [#227](https://github.com/crypto-org-chain/ethermint/pull/227) Reuse sender recovery result.
 * (cli) [#242](https://github.com/crypto-org-chain/ethermint/pull/242) Integrate tendermint bootstrap cmd.
 * (cli) [#246](https://github.com/crypto-org-chain/ethermint/pull/246) Call app.Close to cleanup resource on graceful shutdown.
+* (cli) [#]() make abci handshake shutdown gracefully.
```
Comment (yihuang) [inline CHANGELOG.md]:
```suggestion
* (cli) [#288](https://github.com/crypto-org-chain/ethermint/pull/288) make abci handshake shutdown gracefully.
```

File: go.mod [hunk 2]
```diff
@@ -210,7 +210,7 @@ replace (
 	github.com/miguelmota/go-ethereum-hdwallet => github.com/crypto-org-chain/go-ethereum-hdwallet v0.1.2
 	github.com/syndtr/goleveldb => github.com/syndtr/goleveldb v1.0.1-0.20210819022825-2ae1ddf74ef7
 	// use cometbft
-	github.com/tendermint/tendermint => github.com/cometbft/cometbft v0.34.28
+	github.com/tendermint/tendermint => github.com/yihuang/cometbft v0.34.28-0.20230621082015-c42075bf65d7
```
Comment (yihuang) [inline go.mod]: https://github.com/cometbft/cometbft/pull/1012

File: server/start.go [hunk 3]
```diff
@@ -234,6 +235,7 @@ func startStandAlone(ctx *server.Context, opts StartOptions) error {
 	}
 
 	app := opts.AppCreator(ctx.Logger, db, traceWriter, ctx.Viper)
+	defer app.Close()
```
Comment (mmsqe) [inline server/start.go]: should we handle close error?

Comment (yihuang) [reply server/start.go]: added a log

**Review by mmsqe — APPROVED** [review]
not sure if we could backport [this](https://github.com/cosmos/cosmos-sdk/pull/16252/files) refactor

**Comment by yihuang** [comment]
> not sure if we could backport [this](https://github.com/cosmos/cosmos-sdk/pull/16252/files) refactor  

We can reuse sdk StartCmd in newer version

**Pull request merged**